File: services/news_api/app/models/category.py
"""Category model."""

from sqlalchemy import Boolean, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class Category(Base):
    __tablename__ = "categories"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(32), unique=True, index=True)
    sort_order: Mapped[int] = mapped_column(Integer, default=0)
    enabled: Mapped[bool] = mapped_column(Boolean, default=True)
    include_keywords: Mapped[str] = mapped_column(String(500), default="")
    exclude_keywords: Mapped[str] = mapped_column(String(500), default="")

File: apps/news_app/lib/core/models/category.dart
class CategoryModel {
  CategoryModel({
    required this.id,
    required this.name,
    required this.sortOrder,
    required this.enabled,
    required this.includeKeywords,
    required this.excludeKeywords,
  });

  factory CategoryModel.fromJson(Map<String, dynamic> json) {
    return CategoryModel(
      id: json['id'] as int,
      name: json['name'] as String,
      sortOrder: json['sort_order'] as int? ?? 0,
      enabled: json['enabled'] as bool? ?? true,
      includeKeywords: json['include_keywords'] as String? ?? '',
      excludeKeywords: json['exclude_keywords'] as String? ?? '',
    );
  }

  final int id;
  final String name;
  final int sortOrder;
  final bool enabled;
  final String includeKeywords;
  final String excludeKeywords;
}
